plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.gjs

```diff
@@ -10,48 +11,41 @@ export default class ChatChannelUnreadIndicator extends Component {
 
   get showUnreadIndicator() {
     return (
-      this.args.channel.tracking.unreadCount > 0 ||
-      this.args.channel.tracking.mentionCount > 0 ||
-      this.args.channel.unreadThreadsCountSinceLastViewed > 0
+      this.args.channel.tracking.unreadCount +
+        this.args.channel.tracking.mentionCount +
+        this.args.channel.unreadThreadsCountSinceLastViewed >
+      0
     );
   }
 
-  get urgentCount() {
-    if (this.hasChannelMentions) {
-      return this.args.channel.tracking.mentionCount;
-    }
-    if (this.hasWatchedThreads) {
-      return this.args.channel.tracking.watchedThreadsUnreadCount;
-    }
-    return this.args.channel.tracking.unreadCount;
-  }
-
-  get isUrgent() {
-    if (this.onlyMentions) {
-      return this.hasChannelMentions;
-    }
+  get publicUrgentCount() {
     return (
-      this.isDirectMessage || this.hasChannelMentions || this.hasWatchedThreads
+      this.args.channel.tracking.mentionCount +
+      this.args.channel.tracking.watchedThreadsUnreadCount
     );
   }
 
-  get isDirectMessage() {
+  get directUrgentCount() {
     return (
-      this.args.channel.isDirectMessageChannel &&
-      this.args.channel.tracking.unreadCount > 0
+      this.args.channel.tracking.unreadCount +
+      this.args.channel.tracking.mentionCount +
+      this.args.channel.tracking.watchedThreadsUnreadCount
     );
   }
 
-  get hasChannelMentions() {
-    return this.args.channel.tracking.mentionCount > 0;
+  get urgentCount() {
+    return this.args.channel.isDirectMessageChannel
+      ? this.directUrgentCount
+      : this.publicUrgentCount;
   }
 
-  get hasWatchedThreads() {
-    return this.args.channel.tracking.watchedThreadsUnreadCount > 0;
+  get isUrgent() {
+    return this.urgentCount > 0;
   }
 
-  get onlyMentions() {
-    return hasChatIndicator(this.currentUser).ONLY_MENTIONS;
+  get urgentBadgeCount() {
+    let totalCount = this.urgentCount;
+    return totalCount > MAX_UNREAD_COUNT ? `${MAX_UNREAD_COUNT}+` : totalCount;
```

Comment:
Might want to `i18n`-ize the 99+ string.

Reply:
There are a few places we have the "99+" (header/footer tabs etc) but I don't think we've added any translations for that yet. I think it came up in discussion before but the concern was UX. Perhaps we can do a follow up PR to cover all of these instances.